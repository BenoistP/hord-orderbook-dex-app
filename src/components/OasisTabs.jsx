import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
import { NavLink } from 'react-router-dom';
// import ImmutablePropTypes from 'react-immutable-proptypes';
import './OasisTabs.scss';


const propTypes = PropTypes && {
  children: PropTypes.node
};
const defaultProps = {};


class OasisTabs extends PureComponent {
  render() {
    return (
      <div className="OasisTabs">
        <div className="row-tabs">
          <ul className="tabs">
            <li><NavLink to={'/trade'}>Trade</NavLink></li>
            <li><NavLink to={'/wrap-unwrap'}>Wrap / Unwrap</NavLink></li>
            <li><NavLink to={'/transfer'}>Transfer</NavLink></li>
          </ul>
        </div>
        {this.props.children}
      </div>
    );
  }
}

OasisTabs.displayName = 'OasisTabs';
OasisTabs.propTypes = propTypes;
OasisTabs.defaultProps = defaultProps;
export default OasisTabs;
