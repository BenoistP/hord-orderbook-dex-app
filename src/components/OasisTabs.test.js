/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';

import OasisTabs from './OasisTabs';
import { shallow } from 'enzyme';

describe('(Component) OasisTabs', () => {
  it('should render', () => {
    const props = {};
    const wrapper = shallow(
      <OasisTabs {...props}/>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
