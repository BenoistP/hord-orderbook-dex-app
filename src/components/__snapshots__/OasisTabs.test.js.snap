// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) OasisTabs should render 1`] = `
<div>
  <ul>
    <li
      className="undefined "
    >
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        to="/trade/W-ETH/DAI"
      >
        Trade
      </NavLink>
    </li>
    <li
      className="undefined undefined "
    >
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        to="/wrap-unwrap"
      >
        Wrap / Unwrap
      </NavLink>
    </li>
    <li
      className="undefined "
    >
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        to="/transfer"
      >
        Transfer
      </NavLink>
    </li>
  </ul>
</div>
`;
